web/src/lib/domain-logic/getNavBarBusinessTitle.test.tsx

```diff
@@ -1,141 +1,344 @@
 import { getMergedConfig } from "@/contexts/configContext";
 import { getNavBarBusinessTitle } from "@/lib/domain-logic/getNavBarBusinessTitle";
 import {
+  BusinessPersona,
   generateBusiness,
   LookupIndustryById,
   LookupLegalStructureById,
+  ProfileData,
 } from "@businessnjgovnavigator/shared";
 import { generateProfileData } from "@businessnjgovnavigator/shared/test";
 
 const Config = getMergedConfig();
 
+const name = "some cool name";
+
+const publicFilingProfile = (): Partial<ProfileData> => {
+  return {
+    tradeName: "",
+    legalStructureId: "limited-liability-company",
+  };
+};
+
+const tradeNameProfile = (): Partial<ProfileData> => {
+  return {
+    businessName: "",
+    legalStructureId: "sole-proprietorship",
+  };
+};
+
 describe("getNavBarBusinessTitle", () => {
-  it("shows guest text if industryId is undefined", () => {
-    const business = generateBusiness({
-      profileData: generateProfileData({
-        businessName: "",
-        tradeName: "",
-        industryId: undefined,
-      }),
-    });
-    const navBarBusinessTitle = getNavBarBusinessTitle(business);
-    expect(navBarBusinessTitle).toEqual(Config.navigationDefaults.navBarGuestText);
-  });
+  describe("when name is defined", () => {
+    describe("when legal structure undefined", () => {
+      it.each(["STARTING", "OWNING"])("shows business name", (businessPersona) => {
+        const business = generateBusiness({
+          profileData: generateProfileData({
+            businessPersona: businessPersona as BusinessPersona,
+            businessName: name,
+            tradeName: "",
+            legalStructureId: undefined,
+          }),
+        });
+        const navBarBusinessTitle = getNavBarBusinessTitle(business);
+        expect(navBarBusinessTitle).toEqual(name);
+      });
 
-  it("shows businessName if business name is defined and industryId are also defined", () => {
-    const businessName = "businessName";
-    const business = generateBusiness({
-      profileData: generateProfileData({
-        businessName: businessName,
-        industryId: "cannabis",
-        legalStructureId: "limited-liability-company",
-      }),
+      it.each(["STARTING", "OWNING"])("shows business name", (businessPersona) => {
+        const business = generateBusiness({
+          profileData: generateProfileData({
+            businessPersona: businessPersona as BusinessPersona,
+            businessName: "",
+            tradeName: name,
+            legalStructureId: undefined,
+          }),
+        });
+        const navBarBusinessTitle = getNavBarBusinessTitle(business);
+        expect(navBarBusinessTitle).toEqual(name);
+      });
+
+      it.each(["STARTING", "OWNING"])(
+        "shows business name over trade name if both defined",
```

Comment:
When would this situation occur in our application?

Reply:
If a user switches legal structure. It's not likely but technically possible
1. onboard 
2. set LLC
3. save a business name in profile
4. switch to SP/GP
5. save a trade name in profile
6. remove legal structure altogether